requirements/AUR_CBI_MDL-0230/work/client1/implem/default/C/src/client1.c

```diff
@@ -0,0 +1,97 @@
+/* Body file for function Environment
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "client1.h"
+#include "routing.h"
+
+static asn1SccDataStoreCreateRequest create_request;
+static asn1SccDataStoreDeleteRequest delete_request;
+static asn1SccDataStoreRetrieveRequest retrieve_request;
+static asn1SccDataStoreUpdateRequest update_request;
+
+static int counter;
+static int subscribed;
+
+void client1_startup(void)
+{
+    counter = 0;
+    subscribed = 0;
+}
+
+void client1_PI_Trigger( void )
+{
+    if(subscribed == 0)
+    {
+        asn1SccT_UInt32 event_id = event_id_datastore_notify;
+        asn1SccT_Boolean should_subscribe = 1;
+        client1_RI_subscribe_to_event(&event_id, &should_subscribe);
+        subscribed = 1;
+    }
+    create_request.behaviour = DataStoreCreateRequest_behaviour_reject_when_overflow;
```

Comment:
I think that this test should look a bit different. Let's talk.

Reply:
Fixed